File: python/organizing_policy/list_invitations.py
from organizing_policy.invitation_repository import InvitationRepository


def make_list_invitations(invitation_repository: InvitationRepository):
    def list_invitations(user_id: str, direction: str, status=None) -> list:
        if direction == "sent":
            invitations = invitation_repository.find_by_inviting_user_id(user_id)
        else:
            invitations = invitation_repository.find_by_invited_user_id(user_id)
        if status is not None:
            invitations = [i for i in invitations if i.status == status]
        return invitations
    return list_invitations

File: java/organizing-policy/src/main/java/com/hexagonalchess/organizing/ListInvitations.java
package com.hexagonalchess.organizing;

import java.util.List;
import java.util.stream.Stream;

public class ListInvitations {

    private final InvitationRepository invitationRepository;

    public ListInvitations(InvitationRepository invitationRepository) {
        this.invitationRepository = invitationRepository;
    }

    public List<GameInvitation> execute(String userId, String direction, String status) {
        Stream<GameInvitation> invitations = direction.equals("sent")
            ? invitationRepository.findByInvitingUserId(userId).stream()
            : invitationRepository.findByInvitedUserId(userId).stream();

        if (status != null) {
            var filterStatus = InvitationStatus.valueOf(status.toUpperCase());
            invitations = invitations.filter(i -> i.status() == filterStatus);
        }
        return invitations.toList();
    }
}
